src/x/instrument/extended.go

```diff
@@ -196,7 +203,7 @@ func NewExtendedMetricsReporter(
 	r.init(reportInterval, func() {
 		r.runtime.report(r.metricsType)
 	})
-	if r.metricsType >= ModerateExtendedMetrics {
+	if r.metricsType >= ModerateExtendedMetrics && r.metricsType != DetailedGoRuntimeMetrics {
```

Comment:
not a fan of `const` non-equality comparisons, can we `switch` these to be explicit instead?